lib/lacerda/compare/json_schema.rb

```diff
@@ -111,24 +87,126 @@ def schema_contains?(publish, consume, location = [])
+          #
+          # For each possible object type from the publish schema we have
+          # to check if we find a compatible type in the consume schema.
+          #
+          # It's not sufficient to just compare the names of the objects,
+          # because they might be different in the publish and consume
+          # schemas.
+          elsif publish['oneOf'] and consume['oneOf']
+            publish_types = publish['oneOf']
+            consume_types = [consume['oneOf']].flatten.compact
+
+            # Check all publish types for a compatible consume type
+            publish_types.each do |publish_type|
+              original_errors = @errors
+              @errors = []
+              compatible_consume_type_found = false
+              consume_types.each do |consume_type|
+                next unless publish_type == consume_type or # Because null types are stripped in schema_contains
+                            schema_contains?(publish: publish_type, consume: consume_type, location: location + [publish_type])
+                compatible_consume_type_found = true
+              end
+              @errors = original_errors
+              return _e(:ERR_MISSING_MULTI_PUBLISH_MULTI_CONSUME, location, publish_type) unless compatible_consume_type_found
+            end
+
+          # Mixed case 1/2:
+          elsif consume['oneOf'] and publish['properties']
+            consume_types = ([consume['oneOf']].flatten - [{"type" => "null"}]).sort
+            compatible_consume_type_found = false
+            original_errors = @errors
+            @errors = []
+            consume_types.each do |consume_type|
+              next unless schema_contains?(publish: publish, consume: consume_type, location: location)
+              compatible_consume_type_found = true
+            end
+            @errors = original_errors
+            return _e(:ERR_MISSING_SINGLE_PUBLISH_MULTI_CONSUME, location, publish['type']) unless compatible_consume_type_found
+
+          # Mixed case 2/2:
+          elsif consume['properties'] and publish['oneOf']
+            publish_types = ([publish['oneOf']].flatten - [{"type" => "null"}]).sort
+            incompatible_publish_type= nil
+            original_errors = @errors
+            @errors = []
+            publish_types.each do |publish_type|
+              next if schema_contains?(publish: publish_type, consume: consume, location: location)
+              incompatible_publish_type = publish_type
+            end
+            @errors = original_errors
+            return _e(:ERR_MISSING_MULTI_PUBLISH_SINGLE_CONSUME, location, incompatible_publish_type) if incompatible_publish_type
+
+          # We don't know how to handle this 😳
+          # an object can either have "properties" or "oneOf", if the schema has anything else, we break
+          else
+            return _e(:ERR_NOT_SUPPORTED, location, "Consume schema didn't have properties defined and publish schema no oneOf")

```

Comment:
Reading the json schema spec, this should never happen as object types only have ```properties``` or ```oneOf```.